### PR TITLE
chore: migrate node-local-dns

### PR DESCRIPTION
**Pull request by KraMorK**
## Description

Reduce differences between cse and main branches. Standardize build, update module dependencies
## Why do we need it, and what problem does it solve?

Migrate cse changes to main branch
## Why do we need it in the patch release (if we do)?

not related to release
## What is the expected result?

Migrate cse changes to main branch
## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

```changes
section: node-local-dns
type: chore
summary: Reduce differences between cse and main branches - node-local-dns-module.
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/aws'...

File: ee/be/modules/350-node-local-dns/images/coredns/werf.inc.yaml [hunk 1]
```diff
@@ -1,72 +1,80 @@
...
-  before: setup
-docker:
-  ENTRYPOINT: ["/coredns"]
+  - '**/*.sh'
+  stageDependencies:
+    install:
+    - '**/*.sh'
+shell:
+  install:
+  - git clone --depth 1 --branch v1.11.3 {{ .SOURCE_REPO }}/coredns/coredns.git /src/coredns
```
Comment (name212) [inline ee/be/modules/350-node-local-dns/images/coredns/werf.inc.yaml:17]: please add `rm -rf .git`

**Comment by KraMorK** [comment]
> one thing

fixed. waiting for e2e test and done

**Comment by KraMorK** [comment]
e2e complete, ready for review

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/aws'...

**Comment by KraMorK** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 12045339680 12045339680-1-con-1-27 /sys/deckhouse-oss/install:pr10714 ec2-user@18.199.49.93

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/aws'...

**Review by name212 — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**